File: travel-page-category-web/src/main/java/com/itheima/controller/CategoryController.java
package com.itheima.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.itheima.page.CategoryService;
import com.itheima.pojo.TabCategory;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("category")
public class CategoryController {
    @Reference
    private CategoryService categoryService;

    //查询所有
    @RequestMapping("/findAll")
    public List<TabCategory> findAll() {
        return categoryService.findAll();
    }

    //根据旅游分类id查询
    @RequestMapping("/findById")
    public List<TabCategory> findById(Integer cid) {

        return categoryService.findById(cid);
    }
}

File: travel-page-category-web/src/main/webapp/js/controller/cateGoryController.js
//控制层
app.controller('categoryController',function ($scope,categoryService){
    //读取列表数据绑定到表单中
    $scope.findAll=function(){
        categoryService.findAll().success(
            function (response) {
                $scope.list=response;
            }
        )

    }


/*-------------------------------*/

//根据 ID 显示列表
    $scope.findById=function(cid){
     categoryService.findById(cid).success(
        function(response){
            $scope.list=response;
        }
    );
}

})